corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -292,6 +293,28 @@ def get_attachments(form_id):
         return list(XFormAttachmentSQL.objects.raw('SELECT * from get_form_attachments(%s)', [form_id]))
 
     @staticmethod
+    def iter_forms_by_last_modified(start_datetime, end_datetime):
+        '''
+        Returns all forms that have been modified within a time range. The start date is
+        exclusive while the end date is inclusive (start_datetime, end_datetime].
+
+        NOTE: This does not include archived forms
+
+        :param start_datetime: The start date of which modified forms must be greater than
+        :param end_datetime: The end date of which modified forms must be less than or equal to
+
+        :returns: An iterator of XFormInstanceSQL objects
+        '''
+        from corehq.sql_db.util import run_query_across_partitioned_databases
+
+        return run_query_across_partitioned_databases(
+            XFormInstanceSQL,
+            (Q(received_on__gt=start_datetime) & Q(received_on__lte=end_datetime)) |
+            (Q(deleted_on__gt=start_datetime) & Q(deleted_on__lte=end_datetime)) |
+            (Q(edited_on__gt=start_datetime) & Q(edited_on__lte=end_datetime))
+        )
```

Comment:
I mean that the same for could fall into multiple date ranges

e.g. 
form.received_on = 2017-01-10
form.edited_on = 2017-02-10
form.deleted_on = 2017-04-01

range 1 = 2017-01-01 to 2017-01-31
range 2 = 2017-02-01 to 2017-02-28
range 3 = 2017-04-01 to 2017-04-30